### ISSUE TITLE
last position not being properly updated on death for /backondeath

### ISSUE THREAD

**Issue by mrcoffee1026** [EssentialsX/Essentials#1699]
Teleport, go back... works fine.
Die, go back... goes to last location you teleported to, not the place you died.

EssentialsX version (`/essentials`): 
Essentials 2.0.1-b547

Server software (`/version`): 
Craftbukkit 1.12.2
Server (logs/latest.log):
unneeded, this isn't erroring, it's just not updating the last location on death.
EssentialsX config (if applicable):

**Comment by mdcfe** [comment]
Could you run `/ess debug`, then paste the console output when someone tries to do `/back` after dying?

**Comment by LoXR** [comment]
Think this was reported as well on my server, /back would go back to another location.

**Comment by mrcoffee1026** [comment]
Ok i'm sorry, i need to clarify this issue, apparently.  It's not that /back isn't doing the correct thing, it is?  What's doing the wrong thing is the respawn location when you have backondeath permission.  I'm being told the respawn location after death is somewhat random?  Not sure why that would be

https://pastebin.com/J8bzr7Yu

My recent experience has been to respawn in the last location I had tped to before I'd died last instead of the death location, that... is not very helpful.  I don't know if my configuration needs to be updated for changes in a later build to deal with a change or what.   Death doesn't have to happen, I'm having similar issues just logging back on... and being taken back to the last place I tped to instead of the place I logged out from.  This didn't used to happen so... thinking it may be a recent thing.

**Comment by mdcfe** [comment]
I think I've experienced this before, but I'll see if I can figure out what is causing this (unless someone else is willing to dig deeper).

**Comment by mrcoffee1026** [comment]
I don't know I'm running the b550 now and on all my logoffs today I came back online to right where I left off, so I'm not able to duplicate the issue on demand... but pretty much every time I logged in for days before today I was logging in to some random one of my previous tp locations... just not the spot I logged off at.  I'm hoping the build update magically fixed it somehow but, don't know why, this wasn't one of the recently addressed issues or anything.

**Comment by DrewAMitchell** [comment]
@mrcoffee1026 Sounds like it might have to be put on the backburner until we can get it replicated. I couldn't replicate it myself on b547.

**Comment by mrcoffee1026** [comment]
Yeah sorry.  I can't cause it either right now, and for the moment no one is griping about spawning in random locations.  Also this may not be essentials?  What if it's just Unix... like file access permissions or something, I know it creates files and stores stuff like the last location to them... if - for some reason like a system update - may have caused certain files to be 'locked' then the location would not have been changeable for some time... at least until I rebooted the system, which I have at least twice within the last day or so... that could be a cause... the reboot being the 'solution'.  If there was a database option for this plugin - however - I'd take that to avoid this sort of thing.

**Comment by mdcfe** [comment]
Essentials doesn't write the last teleport location to disk then read it back from the disk - it keeps it in memory while the user is online.

I've encountered this before, albeit without reliable replication steps.

**Comment by mrcoffee1026** [comment]
hrm... yeah I don't know then.  I saw some similar weirdness going on with my /ignore list while testing that command to where it would keep adding the same name over and over to the list... then I'd have to redo it a bunch to get it off the list... then relog and it would be right back on the list (like 20 times)... but I can't swear this doesn't have something to do with using /essentials reload either... seems like every time I do things like /afk stop working... but like hours after the fact.  I don't know if there's like a timed routine that updates the 'live' data or what, but maybe that fails after a reload or something... might explain SOME of the hard to replicate stuff anyway.

**Comment by frizzbee30** [comment]
Just noticed this issue thread, getting similar issues on our test server (and live), looking at our test server, it is indeed the backondeath aspect, that is causing the issue.

**Comment by mrcoffee1026** [comment]
cool?  I suppose.  At least that's a source of the thing anyway.  Also spawn location issue I noticed... something to do with having different spawns for different user groups I think.. MAYBE combined with not having a home named "home"?  Tends to revert you back to the spawn point of the lowest level group on the server instead of your own group's  as a respawn destination.  But I don't know this isn't always true... I had someone kill me in PVP and I respawned on the nether roof for some reason.  I think the only reason I can even get to that is from the /world 1, 2, etc... normally... which no one else has access to but me.  It doesn't seem to be super consistent, but it does seem like the 'back on death' point is anywhere from 1-3 'death' locations behind or possibly just random spawn locations behind your most recent one, making respawn locations seem almost entirely random... I'm sure they're not, I'm just not sure what is giving certain locations priority.
